pkg/agent/apis/types.go

```diff
@@ -216,18 +216,19 @@ func (r ServiceExternalIPInfo) SortRows() bool {
 
 // BGPPolicyResponse describes the response struct of bgppolicy command.
 type BGPPolicyResponse struct {
-	BGPPolicyName string `json:"name,omitempty"`
-	RouterID      string `json:"routerID,omitempty"`
-	LocalASN      int32  `json:"localASN,omitempty"`
-	ListenPort    int32  `json:"listenPort,omitempty"`
+	BGPPolicyName           string `json:"name,omitempty"`
+	RouterID                string `json:"routerID,omitempty"`
+	LocalASN                int32  `json:"localASN,omitempty"`
+	ListenPort              int32  `json:"listenPort,omitempty"`
+	ConfederationIdentifier string `json:"confederationIdentifier,omitempty"`
```

Comment:
I wasn't suggesting changing the type to `string` here. I was suggesting keeping it as `int32` but changing the `GetTableRow` so that it display `""` or `"<NONE>" when the field is `0`. That's fine because 0 is not a valid BGP confederation identified. If you prefer you can make the field an `*int32` and use the ` nil` value, but I don't think that's needed here.